app/controllers/course/assessment/assessments_controller.rb

```diff
@@ -302,6 +399,18 @@ def course_user_ids
     end
   end
 
+  def extract_programming_question_metadata
+    return unless @programming_question.edit_online?
+
+    @meta = programming_package_service.extract_meta
+  end
+
+  def programming_package_service
```

Comment:
why is `Course::Assessment::Question::ProgrammingController` concerns inside assessment controller?

Reply:
moved to KoditsuQuestionConcern